### PR TITLE
Update docs before beta release

### LINKED ISSUE #335: End-user documentation
The following documentation must be provided:
- installation and usage instructions
- troubleshooting (especially the workarounds for assembly version conflicts and missing assemblies, possible telemetry loss)
- supported env (OS and frameworks)
- list of provided instrumentation libraries 
- configuration
- manual instrumentation instructions
- changelog

It would be also good to improve the contributing docs.

### PR DESCRIPTION
**Pull request by Kielek**
Partially fixes #335

Changes proposed in this pull request:

Following cases:

|what | state |
| -- | --|
| installation and usage instructions | TODO: use "releases" instead of CI artifacts |
| list of provided instrumentation libraries | TODO: Reference config.md#instumented-libraries-and-framework in README.md |
| configuration | TODO: Remove all references to MongoDB |

Configuration - all files except examples\README.md - but it is ok to keep it as it is.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -107,6 +107,10 @@ OTEL_DOTNET_AUTO_INTEGRATIONS_FILE=%InstallationLocation%/integrations.json
 
 See [config.md](config.md).
 
+### Instumented libraries and frameworks
+
+See [config.md#instumented-libraries-and-frameworks](config.md#instumented-libraries-and-frameworks).
+
```
Comment (pellared) [inline docs/README.md:113]: How about moving it between the `Compatibility` and `Get Started` sections?

**Review by pellared — APPROVED** [review]

**Review by lachmatt — APPROVED** [review]

**Review by pellared — APPROVED** [review]

File: docs/README.md [hunk 2]
```diff
@@ -37,13 +37,17 @@ CI tests run against the following operating systems:
 - macOS Catalina 10.15,
 - Ubuntu 20.04 LTS.
 
+### Instumented libraries and frameworks
```
Comment (theletterf) [inline docs/README.md:40]:
```suggestion
### Instrumented libraries and frameworks
```

File: docs/config.md [hunk 3]
```diff
@@ -28,20 +28,19 @@ for more details.
...
-| `OTEL_DOTNET_AUTO_{0}_ENABLED` | Configuration pattern for enabling or disabling a specific bytecode. For example, in order to disable MongoDb instrumentation, set `OTEL_TRACE_MongoDb_ENABLED=false` | `true` |
+| `OTEL_DOTNET_AUTO_{0}_ENABLED` | Configuration pattern for enabling or disabling a specific bytecode. For example, in order to disable GraphQL instrumentation, set `OTEL_TRACE_GraphQL_ENABLED=false` | `true` |
 | `OTEL_DOTNET_AUTO_DOMAIN_NEUTRAL_INSTRUMENTATION` |  Sets whether to intercept method calls when the caller method is inside a domain-neutral assembly. This is recommended when instrumenting IIS applications. | `false` |
 | `OTEL_DOTNET_AUTO_CLR_DISABLE_OPTIMIZATIONS` |  Set to `true` to disable all JIT optimizations. | `false` |
 | `OTEL_DOTNET_AUTO_CLR_ENABLE_INLINING` | Set to `false` to disable JIT inlining. | `true` |
 | `OTEL_DOTNET_AUTO_CLR_ENABLE_NGEN` | Set to `false` to disable NGEN images. | `true` |
 
-### Instumented libraries and framework
+### Instumented libraries and frameworks
```
Comment (theletterf) [inline docs/config.md:37]:
```suggestion
### Instrumented libraries and frameworks
```

File: docs/README.md [hunk 4]
```diff
@@ -37,13 +37,17 @@ CI tests run against the following operating systems:
...
+
+See [config.md#instumented-libraries-and-frameworks](config.md#instumented-libraries-and-frameworks).
```
Comment (Kielek) [inline docs/README.md:42]:
```suggestion
See [config.md#instrumented-libraries-and-frameworks](config.md#instumented-libraries-and-frameworks).
```

File: docs/README.md [hunk 5]
```diff
@@ -37,13 +37,17 @@ CI tests run against the following operating systems:
 - macOS Catalina 10.15,
 - Ubuntu 20.04 LTS.
 
+### Instrumented libraries and frameworks
+
+See [config.md#instrumented-libraries-and-frameworks](config.md#instumented-libraries-and-frameworks).
```
Comment (pellared) [inline docs/README.md:42]:
```suggestion
See [config.md#instrumented-libraries-and-frameworks](config.md#instrumented-libraries-and-frameworks).
```

😄

**Review by pellared — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]
LGTM

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**